File: Reqnroll/Chapter18/18.4-SeededDataPattern/Before/WIMP.Specs/Support/Hooks.cs
using Reqnroll;

using WIMP.App.Data;
using WIMP.Specs.Drivers;

namespace WIMP.Specs.Support;

[Binding]
public class Hooks(AppHostingContext appHostingContext, IDatabaseDriver databaseDriver, IDataRepository dataRepository)
{
    [BeforeScenario(Order = 0)]
    public async Task ResetDatabase()
    {
        await databaseDriver.UpgradeSchemaIfNeeded();
        await databaseDriver.EmptyDatabase();
    }

    [BeforeScenario(Order = 1)]
    public void CreateAppHost()
    {
        appHostingContext.AppHost = new WimpAppHost(dataRepository);
    }

    [AfterScenario]
    public void DisposeAppHost()
    {
        appHostingContext.AppHost?.Dispose();
    }
}

File: CucumberJVM/Chapter18/18.3-CleanStartPattern/After/src/test/java/com/wimp/app/specs/support/Hooks.java
package com.wimp.app.specs.support;

import com.wimp.app.specs.drivers.DatabaseDriver;
import io.cucumber.java.*;

public class Hooks {
    private final DatabaseDriver databaseDriver;

    public Hooks(DatabaseDriver databaseDriver) {
        this.databaseDriver = databaseDriver;
    }

    @Before(order = 0)
    public void resetDatabase() {
        databaseDriver.emptyDatabase();
    }
}
